src/renderers/dom/shared/__tests__/ReactDOMComponent-test.js

```diff
@@ -843,6 +842,15 @@ describe('ReactDOMComponent', () => {
       );
     });
 
+    it('should warn if dangerouslySetInnerHTML is not a string', () => {
+      spyOn(console, 'error');
+      mountComponent({dangerouslySetInnerHTML: {__html: {foo: 'bar'} } });
```

Comment:
Can you add tests for other types, like an array, number, and boolean? We should also add a test verifying it doesn't warn when using an object with a custom `toString` implementation.